plugins/modules/packaging/language/ansible_galaxy_install.py

```diff
@@ -0,0 +1,318 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2021, Alexei Znamensky <russoz@gmail.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = """
+module: ansible_galaxy_install
+author:
+    - "Alexei Znamensky (@russoz)"
+short_description: Install Ansible roles or collections using ansible-galaxy
+version_added: 3.5.0
+description:
+  - This module allows the installation of Ansible collections or roles using C(ansible-galaxy).
+notes:
+  - >
+    B(Ansible 2.9/2.10): The C(ansible-galaxy) command changed significantly between Ansible 2.9 and
+    ansible-base 2.10 (later ansible-core 2.11). See comments in the parameters.
+requirements:
+  - Ansible 2.9, ansible-base 2.10, or ansible-core 2.11 or newer
+options:
+  type:
+    description:
+    - The type of installation performed by C(ansible-galaxy).
+    - If I(type) is C(both), then I(requirements_file) must be passed and it may contain both roles and collections.
+    - "Note however that the opposite is not true: if using a I(requirements_file), then I(type) can be any of the three choices."
+    - "B(Ansible 2.9): The option C(both) will have the same effect as C(role)."
+    type: str
+    choices: [collection, role, both]
+    required: true
+  name:
+    description:
+    - Name of the collection or role being installed.
+    - Versions can be specified with C(ansible-galaxy) usual formats. For example, C(community.docker:1.6.1) or C(ansistrano.deploy,3.8.0).
+    - I(name) and I(requirements_file) are mutually exclusive.
+    type: str
+  requirements_file:
```

Comment:
It would be nice to be able to define requirements inline.

Also docs should say if the file is on the target or controller machine.

It would be awesome to ship this module as builtin, to be able to use it for installing `community.general`

Reply:
> It would be nice to be able to define requirements inline.

That is obtained by using the `name` parameter instead of the `requirements_file`, IMO. 

> Also docs should say if the file is on the target or controller machine.

Mixed feelings here: in my mind, everything is on the target except stated otherwise. But I see your point. 

> It would be awesome to ship this module as builtin, to be able to use it for installing `community.general`

For that you will have to argue with the folks in `ansible/ansible`. 😉

Reply:
> > It would be nice to be able to define requirements inline.
> 
> That is obtained by using the `name` parameter instead of the `requirements_file`, IMO.

One could also add a new option `requirements_content` (mutually exclusive, same as the others), whose content is written to a temp file. But I think should be done in a follow-up PR. The current interface is a good MVP to start with.

> > It would be awesome to ship this module as builtin, to be able to use it for installing `community.general`
> 
> For that you will have to argue with the folks in `ansible/ansible`. wink

Ansible (the company) is currently embracing execution environments (EEs), for which such functionality is not needed. So the likelihood of such a module in core is not that high. But I guess it won't hurt to start a discussion on it. Here's definitely the wrong place for that discussion, though.

Reply:
Opened https://github.com/ansible/ansible/issues/75322.

Thanks everyone.

Reply:
Thanks for reviewing @Yajo